samples/ImplicitFlow/AuthorizationServer/Controllers/AuthorizationController.cs

```diff
@@ -0,0 +1,102 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using OpenIddict;
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace AuthorizationServer
+{
+    public class AuthorizationController : Controller
+    {
+        private readonly OpenIddictApplicationManager<OpenIddictApplication<Guid>> _applicationManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly OpenIddictUserManager<ApplicationUser> _userManager;
+
+        public AuthorizationController(
+            OpenIddictApplicationManager<OpenIddictApplication<Guid>> applicationManager,
+            SignInManager<ApplicationUser> signInManager,
+            OpenIddictUserManager<ApplicationUser> userManager)
+        {
+            _applicationManager = applicationManager;
+            _signInManager = signInManager;
+            _userManager = userManager;
+        }
+
+        [Authorize, HttpGet, Route("~/connect/authorize")]
+        public async Task<IActionResult> Authorize()
+        {
+            // Extract the authorization request from the ASP.NET environment.
+            var request = HttpContext.GetOpenIdConnectRequest();
+
+            // Retrieve the application details from the database.
+            var application = await _applicationManager.FindByClientIdAsync(request.ClientId);
+            if (application == null)
+            {
+                return View("Error", new ErrorViewModel
+                {
+                    Error = OpenIdConnectConstants.Errors.InvalidClient,
+                    ErrorDescription = "Details concerning the calling client application cannot be found in the database"
+                });
+            }
+            
+            // Retrieve the profile of the logged in user.
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return View("Error", new ErrorViewModel
+                {
+                    Error = OpenIdConnectConstants.Errors.ServerError,
+                    ErrorDescription = "An internal error has occurred"
+                });
+            }
+
+            // Create a new ClaimsIdentity containing the claims that
+            // will be used to create an id_token, a token or a code.
+            var identity = await _userManager.CreateIdentityAsync(user, request.GetScopes());
+
+            // Create a new authentication ticket holding the user identity.
+            var ticket = new AuthenticationTicket(
+                new ClaimsPrincipal(identity),
+                new AuthenticationProperties(),
+                OpenIdConnectServerDefaults.AuthenticationScheme);
+
+            ticket.SetResources(request.GetResources().Concat(new[] { "ResourceServer01", "ResourceServer02" }));
```

Comment:
`Concat` doesn't seem right. Either hardcode the resources to use `"ResourceServer01"` and `"ResourceServer02"` or use `Intersect`.